[MLOB-7407] Add Automation Rules reference page under monitoring + polish queue/dataset automation tabs (#36432)
DataDog/documentation: content/en/llm_observability/monitoring/automation_rules.md

> ---

Reviewer: this doesn't belong under the `evaluations` folder

Reviewer: as discussed here pls move under "monitoring" here and in the menu that we should also add

https://dd.slack.com/archives/C0658RAV0FL/p1777923364348329?thread_ts=1777920211.040609&cid=C0658RAV0FL

<div class="alert alert-info">Automations apply going forward: new traces matching your rule are routed to the destination as they arrive. Existing traces matching the filter are not added retroactively.</div>

Each rule is evaluated **per span**, in real time, as spans stream into LLM Observability. The rule-matcher does not buffer or join across spans, which constrains the filter set (see [Supported filter fields](#supported-filter-fields)).

## Configuring an automation

You configure automations from the Trace Explorer's **Automate Query** button. The flow is the same regardless of destination:

1. Apply a filter in [**AI Observability > Traces**][1] using [supported filter fields](#supported-filter-fields).
2. Click **Automate Query**.
3. Configure the sampling rate (see [Sampling and limits](#sampling-and-limits)).
4. Pick the destination action:
   - **Add to Annotation Queue**. See [Annotation Queues][2] for queue-specific setup.
   - **Add to Dataset**. See [Datasets][3] for dataset-specific setup.

Manage your rules from [**AI Observability > Settings > Automations**][4]. You can enable, disable, edit, or delete rules at any time.

> ## Supported filter fields

Author: source of truth - BE (field_accessor.go / field_mapping.go in the rule-matcher repo)

Author: and automation-allowed-fields.constants.ts in web-ui

Each rule is evaluated **per span**, in real time, as spans stream into LLM Observability. The rule-matcher does not buffer or join across spans, which constrains the filter set (see [Supported filter fields](#supported-filter-fields)).

## Configuring an automation

You configure automations from the Trace Explorer's **Automate Query** button. The flow is the same regardless of destination:

1. Apply a filter in [**AI Observability > Traces**][1] using [supported filter fields](#supported-filter-fields).
2. Click **Automate Query**.
3. Configure the sampling rate (see [Sampling and limits](#sampling-and-limits)).
4. Pick the destination action:
   - **Add to Annotation Queue**. See [Annotation Queues][2] for queue-specific setup.
   - **Add to Dataset**. See [Datasets][3] for dataset-specific setup.

Manage your rules from [**AI Observability > Settings > Automations**][4]. You can enable, disable, edit, or delete rules at any time.

## Supported filter fields

Automation filters use a subset of the Trace Explorer search syntax. When a filter contains an unsupported field, the **Automate Query** button is disabled and the tooltip names the offending field.

> | Filter for | Fields |

Reviewer: we should make it clear that these (e.g. `@meta.error.stack`, `@meta.input.prompt.*`, etc) are examples, i don't think we'll do a good job keeping this list up to date

| Model and span kind | `@model_name`, `@model_provider`, `@meta.span.kind` |
| Evaluations | `@evaluation.<NAME>.value`, `@evaluation.<NAME>.assessment`, `@evaluation.<NAME>.reasoning`, `@evaluation.<NAME>.metadata`, `@evaluation.<NAME>.tags` |
| Metrics | `@metrics.*` (any custom metric, for example, `@metrics.input_tokens`) |
| Tags | bare keys (`env:prod`), `@tags` |
| Custom metadata | `@meta.metadata.*` (any key, for example, `@meta.metadata.user_id`) |
| Prompts and tools | `@meta.input.prompt.*`, `@meta.output.prompt.*`, `@meta.tool_definitions.*` |

## Sampling and limits

Automations include a sampling rate to keep volume manageable.

| Destination | Sampling rate cap | Record cap |
|---|---|---|
| Annotation queue | Up to 5% | 1,000 records per queue. The automation pauses when the queue reaches the limit. |
| Dataset | Up to 100% | 20,000 records per dataset. Datasets populated by automations are read-only to prevent accidental modification. |

A higher sampling rate captures more traces but may overflow the destination faster. Start low (1-2%) for queues and tune up as you observe real volume.

## Troubleshooting

> **My rule is saved but no traces are showing up.**

Reviewer 2: ```suggestion
**No traces appear after saving a rule.**
```

Most often, one of:

- **Forward-only**: the rule only acts on traces arriving after you saved it. Wait for new traces, or check that traffic actually matches the filter.
- **Sampling**: a 1% sampling rate routes 1 in 100 matching traces. Confirm the rate is high enough for your match volume.
- **Destination is full**: annotation queues cap at 1,000 records and pause; datasets cap at 20,000.

> **My filter validates in the Trace Explorer but the Automate Query button is disabled.**

Reviewer 2: ```suggestion
**The Automate Query button is disabled despite a valid Trace Explorer filter.**
```
